Bigtable/tests/Unit/DataClientTest.php

```diff
@@ -0,0 +1,227 @@
+        $rowMutation = new RowMutation('rk1');
+        $rowMutation->upsert('cf1', 'cq1', 'value1');
+        $this->entries[] = $rowMutation->getEntry();
+        $this->rowMutations[] = $rowMutation;
+
+        $rowMutation = new RowMutation('rk2');
+        $rowMutation->upsert('cf2', 'cq2', 'value2');
+        $this->entries[] = $rowMutation->getEntry();
+        $this->rowMutations[] = $rowMutation;
+    }
+
+    public function testMutateRows()
+    {
+        $statuses = [];
+        foreach ($this->rowMutations as $rowMutation) {
+            $status = new Status;
+            $status->setCode(Code::OK);
+            $statuses[] = $status;
+        }
+        $mutateRowsResponses = $this->getMutateRowsResponse($statuses);
+        $this->serverStream->readAll()
+            ->shouldBeCalled()
+            ->willReturn(
+                $this->arrayAsGenerator($mutateRowsResponses)
+            );
+        $this->bigtableClient->mutateRows(self::TABLE_NAME, $this->entries, $this->options)
+            ->shouldBeCalled()
+            ->willReturn(
+                $this->serverStream->reveal()
+            );
+        $this->dataClient->mutateRows($this->rowMutations);
+    }
+
+    /**
+     * @expectedException Google\Cloud\Bigtable\Exception\BigtableDataOperationException
+     * @expectedExceptionMessage Invalid argument
+     */
+    public function testMutateRowsFailure()
+    {
+        $statuses = [];
+        $status = new Status;
+        $status->setCode(Code::INVALID_ARGUMENT);
+        $status->setMessage('Invalid argument');
+        $statuses[] = $status;
+        $status = new Status;
+        $status->setCode(Code::OK);
+        $statuses[] = $status;
+
+        $mutateRowsResponses = $this->getMutateRowsResponse($statuses);
+        $this->serverStream->readAll()
+            ->shouldBeCalled()
+            ->willReturn(
+                $this->arrayAsGenerator($mutateRowsResponses)
+            );
+        $this->bigtableClient->mutateRows(self::TABLE_NAME, $this->entries, $this->options)
+            ->shouldBeCalled()
+            ->willReturn(
+                $this->serverStream->reveal()
+            );
+        $this->dataClient->mutateRows($this->rowMutations);
+    }
+
+    /**
+     * @expectedException Google\ApiCore\ApiException
+     * @expectedExceptionMessage unauthenticated
+     */
+    public function testMutateRowsApiExceptionInMutateRows()
+    {
+        $apiException =  new ApiException('unauthenticated', Code::UNAUTHENTICATED, 'unauthenticated');
+        $this->bigtableClient->mutateRows(self::TABLE_NAME, $this->entries, $this->options)
+            ->shouldBeCalled()
+            ->willThrow(
+                $apiException
+            );
+        $this->dataClient->mutateRows($this->rowMutations);
+    }
+
+    /**
+     * @expectedException Google\ApiCore\ApiException
+     * @expectedExceptionMessage unauthenticated
+     */
+    public function testMutateRowsApiExceptionInreadAll()
```

Comment:
Nit: You need to capitalize "Read" too: testMutateRowsApiExceptionInreadAll

Reply:
done